pkg/api/customization/project/project_store.go

```diff
@@ -171,12 +175,51 @@ func isQuotaFit(apiContext *types.APIContext, nsQuotaLimit *v3.ResourceQuotaLimi
 		return err
 	}
 
-	if project.ResourceQuota == nil || project.ResourceQuota.UsedLimit == nil {
+	if project.ResourceQuota == nil {
 		return nil
 	}
 
+	// check if fields were added or removed
+	// and update project's namespaces accordingly
+	defaultQuotaLimitMap, err := convert.EncodeToMap(nsQuotaLimit)
+	if err != nil {
+		return err
+	}
+
+	usedQuotaLimitMap := map[string]interface{}{}
+	if project.ResourceQuota.UsedLimit != nil {
+		usedQuotaLimitMap, err = convert.EncodeToMap(project.ResourceQuota.UsedLimit)
+		if err != nil {
+			return err
+		}
+	}
+
+	limitToAdd := map[string]interface{}{}
+	limitToRemove := map[string]interface{}{}
+	for key, value := range defaultQuotaLimitMap {
+		if _, ok := usedQuotaLimitMap[key]; !ok {
+			limitToAdd[key] = value
+		}
+	}
+
+	for key, value := range usedQuotaLimitMap {
+		if _, ok := defaultQuotaLimitMap[key]; !ok {
+			limitToRemove[key] = value
+		}
+	}
+
 	// check that used quota is not bigger than the project quota
-	usedQuotaLimit, err := limitToLimit(project.ResourceQuota.UsedLimit)
+	for key := range limitToRemove {
+		delete(usedQuotaLimitMap, key)
+	}
+
+	var usedLimitToCheck mgmtclient.ResourceQuotaLimit
+	err = convert.ToObj(usedQuotaLimitMap, &usedLimitToCheck)
+	if err != nil {
+		return err
+	}
+
+	usedQuotaLimit, err := limitToLimit(&usedLimitToCheck)
```

Comment:
not sure if we need to convert this again, since it's already present in `project.ResourceQuota.UsedLimit` ?

Reply:
@kinarashah oh the fun with types. Objects fetched using access.ByID would have different types from if you'd access them directly using API. As the isQuotaFit meant to be used by both controller and api layer, the conversion to the type the controller would understand too, is necessary